pkg/local_object_storage/engine/tree.go

```diff
@@ -0,0 +1,78 @@
+package engine
+
+import (
+	"github.com/nspcc-dev/neofs-node/pkg/local_object_storage/pilorama"
+	cidSDK "github.com/nspcc-dev/neofs-sdk-go/container/id"
+)
+
+var _ pilorama.Forest = (*StorageEngine)(nil)
+
+// TreeMove implements the pilorama.Forest interface.
+func (e *StorageEngine) TreeMove(cid *cidSDK.ID, treeID string, m *pilorama.Move) (*pilorama.LogMove, error) {
+	var err error
+	var lm *pilorama.LogMove
+	for _, sh := range e.sortShardsByWeight(cid) {
```

Comment:
We can also use `iterateOverSortedShards` here.

Reply:
It needs an address, we only have CID, and there are no gains in terms of code size, so I decided to write a for loop

Reply:
Actually we can change parameter type to `address.Address` -> `fmt.Stringer`. But I don't mind to leave for-loop.